Lib/asyncio/base_events.py

```diff
@@ -1866,19 +1866,19 @@ def _run_once(self):
             handle = self._ready.popleft()
             if handle._cancelled:
                 continue
-            if self._debug:
-                try:
-                    self._current_handle = handle
-                    t0 = self.time()
-                    handle._run()
-                    dt = self.time() - t0
-                    if dt >= self.slow_callback_duration:
-                        logger.warning('Executing %s took %.3f seconds',
-                                       _format_handle(handle), dt)
-                finally:
-                    self._current_handle = None
-            else:
+            try:
+                self._current_handle = handle
+                t0 = self.time()
```

Comment:
I'm not familiar enough to know whether these are okay to do even when not debugging, particularly the `_current_handle` assignment.

Reply:
The report should be in debug mode only, sure.
Two timer calls per callback are not free.

Reply:
Is there a better way of detecting a stall? I did a quick bit of bechmarking and a raw, non-lookup'd call to `time.time()` or `time.monotonic()` is about as fast as you can get (~50ns), on-par with invoking `lambda: None` or `def foo(): pass`.

A 50ns overhead leaves a _lot_ of breathing room.